crates/oxc_semantic/src/binder.rs

```diff
@@ -96,7 +96,9 @@ impl<'a> Binder for Function<'a> {
             if !scope.strict_mode && matches!(builder.parent_kind(), AstKind::IfStatement(_)) {
                 // Do not declare in if single statements,
                 // if (false) function f() {} else function g() { }
-            } else if self.r#type == FunctionType::FunctionDeclaration {
+            } else if self.r#type == FunctionType::FunctionDeclaration
+                || self.r#type == FunctionType::FunctionExpression
```

Comment:
this is for 
```js
var a = function foo() { foo = 123; };
```
But I'm not sure that's the right change.

Reply:
Can you run `cargo coverage` and see if it breaks anything

Reply:
You can also comment out this test case if it's not that easy to implement.